ethash_opencl.go

```diff
@@ -560,7 +560,7 @@ func (c *OpenCLMiner) Search(block pow.Block, stop <-chan struct{}, index int) (
 					ds := C.uint64_t(c.dagSize)
 					// We verify that the nonce is indeed a solution by
 					// executing the Ethash verification function (on the CPU).
-					ret := C.ethash_light_compute_internal(c.ethash.Light.current.ptr, ds, hashToH256(headerHash), cn)
+					ret := C.ethash_light_compute_internal(c.ethash.Light.getCache(block.NumberU64()).ptr, ds, hashToH256(headerHash), cn)
```

Comment:
You can make a method on Light that does this check and then use it here and in Light.Verify.
Basically refactor the core of Light.Verify into its own method.